blockstore/apps/bundles/storage.py

```diff
@@ -0,0 +1,165 @@
+
+        The first set of credentials remain the default credentials for Blockstore,
+        while the second set of credentials can be specifically used to sign
+        S3 URLs. Hence, this class.
+
+    THIS CLASS IS SPECIFIC TO THE S3Boto3STorage BACKEND.
+    It will not work with FileSystemStorage or any other backend.
+    (If it seemed prudent, though, this class could be generalized to work
+     with other storage backends.)
+    """
+
+    class BackendNotAvailable(Exception):
+        """
+        Raised if `LongLivedSignedUrlStorage` is not configured for use.
+        """
+        def __str__(self):
+            return (
+                f"In order to instantiate this storage backend, "
+                "boto3 must be installed "
+                "and both BUNDLE_ASSET_URL_STORAGE_KEY "
+                "and BUNDLE_ASSET_URL_STORAGE_SECRET "
+                "must be configured in Django settings."
+            )
+
+    def __init__(self):
+        """
+        Construct a storage backend instance.
+
+        Raises `BackendNotAvailable` if we are not configured.
+        """
+        try:
+            S3Boto3Storage = get_storage_class("storages.backends.s3boto3.S3Boto3Storage")
+        except ImportError as import_error:
+            raise self.BackendNotAvailable from import_error
+        try:
+            key = settings.BUNDLE_ASSET_URL_STORAGE_KEY
+            secret = settings.BUNDLE_ASSET_URL_STORAGE_SECRET
+        except AttributeError as attr_error:
+            raise self.BackendNotAvailable from attr_error
+        if not (key and secret):
+            raise self.BackendNotAvailable
+        self.s3_backend = S3Boto3Storage(
+            # All other S3 settings will be pulled in automatically from Django settings
+            # (such as AWS_QUERYSTRING_EXPIRE and AWS_STORAGE_BUCKET_NAME).
+            AWS_ACCESS_KEY_ID=key,
+            AWS_SECRET_ACCESS_KEY=secret,
+        )
+
+    def url(self, name):
+        """
+        Generate a URL using the configured backend.
+        """
+        return self.s3_backend.url(name)
+
+
+class AssetStorage(Storage):
+    """
+    Storage backend for assets related to bundles.
+
+    This is largely a pass-through wrapper around the `default_storage` class;
+    the one key difference is that URLs are generated through the
+    `LongLivedSignedUrlStorage` class, if it is active.
+    """
+
+    def __init__(self):
+        """
+        Initialize an instance of AssetStorage.
+
+        If `LongLivedSignedUrlStorage` is active, then instantiate an instance of
+        it for generating URLs; otherwise, fall back to the default storage class.
+        """
+        try:
+            self.url_backend = LongLivedSignedUrlStorage()
+        except LongLivedSignedUrlStorage.BackendNotAvailable:
+            self.url_backend = default_storage
+
+    def url(self, name):
+        return self.url_backend.url(name)
+
+    def delete(self, name):
```

Comment:
I'm not in love with the long list of pass-through methods I needed to put here (I used [this page](https://docs.djangoproject.com/en/3.1/howto/custom-file-storage/) and pylint's "abstract method warnings" as a guide), but it seemed more straightforward than making the class a bona fide proxy with `__getattr__` magic. I can revisit this if reviewers would prefer it.